test/client.spec.ts

```diff
@@ -123,10 +145,47 @@ describe('Certificate Client', () => {
 					return Bluebird.fromCallback<string>(cb => {
 						return DNS.lookup(uri, cb);
 					}).then((ipAddress: string) => {
-						ipAddress.should.equal(testIP);
+						ipAddress.should.equal(testIp);
 					});
 				},
 			);
 		});
+
+		// Note that we don't test for the updated IP address here. This is because
+		// we'd need to clear the DNS cache first, which is OS specific and also could
+		// affect it's immediate operation. We take the successful passing of this test to
+		// imply the A record has been correctly updated.
+		it(
+			'should not regenerate a certificate when a pre-existing one exist, but should ' +
+				'update DNS A records',
+			() => {
+				const client = new BalenaCertificateClient({
+					dnsUpdateHost: dnsHost,
+					dnsUpdatePort: dnsPort,
+					authToken: process.env.CERT_TEST_AUTH_TOKEN || '',
+					configRoot: `${process.cwd()}/test/testcert`,
+				});
+
+				return client
+					.requestCertificate({
+						domain,
+						subdomains: [`*.${uuid}`, `*.devices.${uuid}`],
+						ip: updatedIp,
+						email,
+						renewing: true,
+					})
+					.then(_result => {
+						return new Error('New certificate should not have been generated');
```

Comment:
`throw new Error`

And below.

Reply:
Must have really been having a bad day....